### ISSUE TITLE
Segment Filter Not Working 2.15.1 April 2019

### ISSUE THREAD

**Issue by ankush1492** [mautic/mautic#7485]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
| Q   | A
| --- | ---
| Bug Report | X
| Feature Request | ---

| Q   | A
| --- | ---
| Mautic Version | 2.15.1  
| PHP Version | 7.2 

### Steps to reproduce
1. Create a segment.
2. Add filter and save.
 
Tried on different installations. Same issue
### Log errors

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by carloshmm** [comment]
Could you please post the logs from your Mautic environment (if any exist).
Also, have you tried to execute the cronjob segments:update (https://www.mautic.org/docs/en/setup/cron_jobs.html)? Did it show any log that could help to solve your problem?

**Comment by ankush1492** [comment]
There are no logs. Logs are empty. And cron is setup properly. Also, while setting up the filters for already existing contacts does cron jobs affect them?

**Comment by escopecz** [comment]
Please provide more info. What happens when you add filter and save? Can you add a screenshot?

**Comment by ankush1492** [comment]
![filter save](https://user-images.githubusercontent.com/18229441/56947886-1a091480-6b4c-11e9-8a55-0ec679473d7a.jpg)
![no_contatcs](https://user-images.githubusercontent.com/18229441/56947887-1a091480-6b4c-11e9-9208-8c6321840b69.jpg)

**Comment by escopecz** [comment]
Please, pretty please, help us understand what is the problem!

Is it that there are contacts that match the filters but the segment does not contain them?

**Comment by ankush1492** [comment]
> Is it that there are contacts that match the filters but the segment does not contain them?

Yes, I have 10 contacts wich HAVE the email addresses in them. And when I create a segment with a filter, "Email ID -> not empty" and save it. And then I go to the segment which I created (here by the name of 'test') it shows no contacts. I have tried with First Name -> not empty and also by adding a name ex. Ankush and then putting a filter First Name->equals->Ankush then also no contacts show up in the segment.

**Comment by escopecz** [comment]
Thanks you! It's clear now. Looks like you missed step 2 in the getting started guide: https://www.mautic.org/docs/en/setup/getting_started.html

**Comment by ankush1492** [comment]
OK, I already added cron jobs and still getting the problem. I will confirm if I am putting the correct path. Also, is there any way I can check if the cron jobs are working or not?

**Comment by escopecz** [comment]
Good thinking. The simplest way to find out is to execute the command you put to the crontab via SSH and see what the result is. Another option is to configure logging the cron job results. As it can be different for each platform please use Google to find a tutorial on how to do that.

**Comment by ankush1492** [comment]
OK, great. I will confirm my cron jobs are working if it still reproduces the problem or not.

**Comment by ankush1492** [comment]
As it turns out the path was not set up properly on cron jobs. Now when I have corrected the path. Segment shows the contacts. Thanx

**Comment by Dancpa** [comment]
error Mautic v2.16.0 does not update the segments


Hello to the whole github community, I wanted to comment that I have just installed Mautic v2.16.0 and I have a problem in the segments part, I have created a test tag and when I re-register a contact in a form my contacts are not update in the segment that eleji.

He configured everything and only that problem I present, if someone can help or favor me, thank you very much for your help.

I tried executing the cron job and effectively only this way they are updated.
